### PR TITLE
CCM-7747: Remove DOB

### PR DESCRIPTION
**Pull request by simonlabarere**
## Summary

Remove DOB validation

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Checklist
* [ ] Brief description of work completed, and any technical decisions made as part of the PR
* [ ] PR link added as a comment to the relevant JIRA ticket
* [ ] PR link shared on Slack and/or Teams
* [ ] 2 reviews received
* [ ] Tester approval

### CODE REVIEW

File: tests/development/message_batches/create_message_batches/test_success.py [hunk 1]
```diff
@@ -106,45 +106,8 @@ def test_201_message_batch_valid_contact_details(
...
-
-    resp = requests.post(
-        f"{nhsd_apim_proxy_url}{MESSAGE_BATCHES_ENDPOINT}",
-        headers={
-            "Authorization": bearer_token_internal_dev.value,
-            "Accept": constants.DEFAULT_CONTENT_TYPE,
-            "Content-Type": constants.DEFAULT_CONTENT_TYPE,
-        },
-        json=data,
+    Assertions.assert_201_response(
```
Comment (aaron-downing) [inline tests/development/message_batches/create_message_batches/test_success.py:109]: I'm guessing this needs the latest main pulling in?

**Review by aaron-downing — APPROVED** [review]

File: tests/development/message_batches/create_message_batches/test_success.py (hunk 1 above)
Comment (simonlabarere) [reply tests/development/message_batches/create_message_batches/test_success.py]: Yep, done!

**Comment by adadigital** [comment]
I found references to dateOfBirth still in `generate_valid_create_message_batch_body` in `tests/lib/generators.py` on lines 29 and 40. Does anything depend on these?

File: proxies/shared/resources/jsc/helpers/validationChecks.js [hunk 2]
```diff
@@ -106,21 +106,6 @@ const validateArray = (errors, fieldValue, fieldPointer, minElements) => {
     return true
 }
 
-const validateDob = (errors, fieldValue, fieldPointer) => {
-    if (
-        !isUndefined(fieldValue)
-        && (
-            fieldValue === null
-            || typeof fieldValue !== "string"
-            || !dobRegex.test(fieldValue)
```
Comment (tdroza-nhs) [inline proxies/shared/resources/jsc/helpers/validationChecks.js:115]: Looks like you can also remove line 4 in this file?
```
const dobRegex = /^\d{4}-\d{2}-\d{2}$/i
```

**Review by tdroza-nhs — CHANGES_REQUESTED** [review]

File: proxies/shared/resources/jsc/helpers/validationChecks.js (hunk 2 above)
Comment (simonlabarere) [reply proxies/shared/resources/jsc/helpers/validationChecks.js:115]: Nice catch! Done.

**Review by tdroza-nhs — APPROVED** [review]

**Comment by simonlabarere** [comment]
> sandbox/messages/2WL4g2f7GQ2F9HZmgonIFiXAPku.json
It looks like this was only referenced in the sandbox postman collection so I removed it from there as well.

> I found references to dateOfBirth still in `generate_valid_create_message_batch_body` in `tests/lib/generators.py` on lines 29 and 40. Does anything depend on these?
Those are now removed as well.

**Review by adadigital — APPROVED** [review]

**Pull request merged**